--- origsrc/libmowgli-0.7.0/m4/buildsys.m4	2008-07-06 08:48:26.000000000 -0500
+++ src/libmowgli-0.7.0/m4/buildsys.m4	2008-07-06 23:30:03.890625000 -0500
@@ -66,7 +66,22 @@
 			UNINSTALL_LIB='rm -f ${DESTDIR}${libdir}/$$i ${DESTDIR}${libdir}/$$i.${LIB_MAJOR}.${LIB_MINOR}'
 			CLEAN_LIB=''
 			;;
-		*-*-cygwin | *-*-mingw32)
+		*-*-cygwin)
+			AC_MSG_RESULT(Cygwin)
+			LIB_CPPFLAGS='-DPIC'
+			LIB_CFLAGS=''
+			LIB_LDFLAGS='-shared -Wl,--enable-auto-image-base,--out-implib,lib${LIB_NAME}.dll.a'
+			LIB_PREFIX='cyg'
+			LIB_SUFFIX='-${LIB_MAJOR}.dll'
+			PLUGIN_CPPFLAGS=''
+			PLUGIN_CFLAGS=''
+			PLUGIN_LDFLAGS='-shared -Wl,--enable-auto-image-base'
+			PLUGIN_SUFFIX='.dll'
+			INSTALL_LIB='${MKDIR_P} ${DESTDIR}${bindir} && ${INSTALL} -m 755 $$i ${DESTDIR}${bindir}/$$i && ${INSTALL} -m 755 lib${LIB_NAME}.dll.a ${DESTDIR}${libdir}/lib${LIB_NAME}.dll.a'
+			UNINSTALL_LIB='rm -f ${DESTDIR}${bindir}/$$i ${DESTDIR}${libdir}/lib${LIB_NAME}.dll.a'
+			CLEAN_LIB='lib${LIB_NAME}.dll.a'
+			;;
+		*-*-mingw32)
 			AC_MSG_RESULT(Win32)
 			LIB_CPPFLAGS='-DPIC'
 			LIB_CFLAGS=''
--- origsrc/libmowgli-0.7.0/src/libmowgli/Makefile	2008-07-06 08:48:26.000000000 -0500
+++ src/libmowgli-0.7.0/src/libmowgli/Makefile	2008-07-06 23:26:57.093750000 -0500
@@ -1,4 +1,5 @@
-LIB = ${LIB_PREFIX}mowgli${LIB_SUFFIX}
+LIB_NAME=mowgli
+LIB = ${LIB_PREFIX}${LIB_NAME}${LIB_SUFFIX}
 LIB_MAJOR = 2
 LIB_MINOR = 0
 DISTCLEAN = mowgli_config.h
